agent/app/agent_capability_test.go

```diff
@@ -162,6 +163,13 @@ func TestCapabilities(t *testing.T) {
 }
 
 func TestCapabilitiesECR(t *testing.T) {
+	pathExists = func(path string, shouldBeDirectory bool) (bool, error) {
```

Comment:
Another option to reduce verbosity would be to create a function like below:

```golang
func mockPathExists() func() {
	pathExists = func(path string, shouldBeDirectory bool) (bool, error) {
		return false, nil
	}
	return func() {
		pathExists = defaultPathExists
	}()
}
```

And invoke it here like so: `defer mockPathExists()()`

Reply:
change to add `mockPathExists()`, `getSubDirectories` is only mocked in `TestCapabilities` because we want to enable `execute-command` capability and verify that it exists so not adding helper function for it